### ISSUE TITLE
Do not insert a newline after function call

### ISSUE THREAD

**Issue by Mic92** [nix-community/nixpkgs-fmt#203]
## Input
```nix
{ hello, fetchurl }:

hello.overrideDerivation (old: {
  src = fetchurl {
    url = "https://example.com/hello-2.6.1";
  };
})
```

## Output
```nix
{ hello, fetchurl }:

hello.overrideDerivation
  (old: {
    src = fetchurl {
      url = "https://example.com/hello-2.6.1";
    };
  })
```

## Desired output

```nix
{ hello, fetchurl }:

hello.overrideDerivation (old: {
  src = fetchurl {
    url = "https://example.com/hello-2.6.1";
  };
})
```

The above formatting in nixpkgs is quite common and consistent with other function calls such as 

```
stdenv.mkDerivation {
  src = fetchurl {
    url = "https://example.com/hello-2.6.1";
  };
}
```

**Comment by Rizary** [comment]
Thank you.

This is happened because wewant to make this: https://github.com/nix-community/nixpkgs-fmt/blob/master/test_data/nixpkgs_repository/maintainers_scripts_update.good.nix readable. you can find it in nixpkgs repo [here](https://github.com/NixOS/nixpkgs/blob/master/maintainers/scripts/update.nix#L47-L65) and [here](https://github.com/NixOS/nixpkgs/blob/master/maintainers/scripts/update.nix#L74-L76). I also notice in some nixpkgs' file, they put the second and third argument align with the function call like:
```
hello.fn (old: {
  src = fetchurl {
    url = "https://example.com/hello-2.6.1";
  };
})
foo
bar
```

that's why I insert newline to make it like:
```
hello.fn 
  (old: {
    src = fetchurl {
      url = "https://example.com/hello-2.6.1";
    };
  })
  foo
  bar
```

otherwise, I will have:

lign with the function call like:
```
hello.fn (old: {
  src = fetchurl {
    url = "https://example.com/hello-2.6.1";
  };
})
  foo
  bar
```

I'm aware of this, and still thinking what is the best approach to format function application. Currently, my plan is to make `(` in newline if the function argument is more than 1.

**Comment by Mic92** [comment]
Tricky. Are there more examples like this?

**Comment by Rizary** [comment]
Yep, like in here: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/config/krb5/default.nix#L11-L13.

The current formatter will look like:
```
libdefaults = optionalAttrs (cfg.defaultRealm != null)
{ default_realm = cfg.defaultRealm; };
```

**Comment by Mic92** [comment]
Would be the line length a good heuristic?

**Comment by Mic92** [comment]
Actually the newline in this example is also not necessary:

```
libdefaults = optionalAttrs (cfg.defaultRealm != null) { 
  default_realm = cfg.defaultRealm; 
};
```

I would say for `optional` we often put the parentheses/brackets on the same line.

**Comment by zimbatm** [comment]
We currently don't enforce a line length limit. It would probably explode the tools' complexity.

**Comment by Rizary** [comment]
> Actually the newline in this example is also not necessary:
> 
> ```
> libdefaults = optionalAttrs (cfg.defaultRealm != null) { 
>   default_realm = cfg.defaultRealm; 
> };
> ```
> 
> I would say for `optional` we often put the parentheses/brackets on the same line.

Agreed. But nixpkgs-fmt will not format by wrapping the code assuming the programmer "does" want that style. So if they put it in line, it will not get expanded too.

Thus, the determination is based on the input file. That's why we can expand or wrap the code in certain cases, but not for all. 

The solution is to make sure the indentation of the code is correct so it can be readable.

**Comment by Rizary** [comment]
In this case, what I want to do is just expand it because it contains newline. So the code:

```
libdefaults = optionalAttrs (cfg.defaultRealm != null) { 
  default_realm = cfg.defaultRealm; 
};
```

Will become:
```
libdefaults = 
  optionalAttrs 
    (cfg.defaultRealm != null) 
    { 
      default_realm = cfg.defaultRealm; 
    };
```
Thus, the programmers

**Comment by Rizary** [comment]
fixed in #204

**Comment by Mic92** [comment]
Thanks!
